ndicapi.cxx

```diff
@@ -1740,7 +1740,7 @@ namespace
         api->BxPortStatus[i] = (int)replyIndex[0] | (int)replyIndex[1] << 8 | (int)replyIndex[2] << 16 | (int)replyIndex[3] << 24;
         replyIndex += 4;
         // 4 bytes frame number
-        api->BxFrameNumber[i] = (unsigned int)replyIndex[0] | (unsigned int)replyIndex[1] << 8 | (unsigned int)replyIndex[2] << 16 | (unsigned int)replyIndex[3] << 24;
+        api->BxFrameNumber[i] = (unsigned int)(unsigned char)replyIndex[0] | (unsigned int)(unsigned char)replyIndex[1] << 8 | (unsigned int)(unsigned char)replyIndex[2] << 16 | (unsigned int)(unsigned char)replyIndex[3] << 24;
```

Comment:
Is the double casting necessary? What happens if you just cast to unsigned char?

Reply:
Ah, you're right, that works. I guess the bitshift performs the cast anyways.